Sources/Commands/SwiftPackageCollectionsTool.swift

```diff
@@ -0,0 +1,358 @@
+        static let configuration = CommandConfiguration(abstract: "Refresh configured collections")
+
+        @OptionGroup
+        var profileOptions: ProfileOptions
+
+        mutating func run() throws {
+            let collections = try await { self.collections.refreshCollections(in: profileOptions.usedProfile, callback: $0) }
+            print("Refreshed \(collections.count) configured package collections.")
+        }
+    }
+
+    struct Add: ParsableCommand {
+        static let configuration = CommandConfiguration(abstract: "Add a new collection")
+
+        @Argument(help: "URL of the collection to add")
+        var collectionUrl: String?
+
+        @Option(name: .long, help: "Sort order for the added collection")
+        var order: Int?
+
+        @OptionGroup
+        var profileOptions: ProfileOptions
+
+        mutating func run() throws {
+            guard let collectionUrlString = collectionUrl, let collectionUrl = URL(string: collectionUrlString) else {
+                throw CollectionsError.missingArgument("collectionUrl")
+            }
+
+            let source = PackageCollectionsModel.CollectionSource(url: collectionUrl)
+            let collection = try await { self.collections.addCollection(source, order: order, to: profileOptions.usedProfile, callback: $0) }
+
+            print("Added \"\(collection.name)\" to your package collections.")
+        }
+    }
+
+    struct Remove: ParsableCommand {
+        static let configuration = CommandConfiguration(abstract: "Remove a configured collection")
+
+        @OptionGroup
+        var profileOptions: ProfileOptions
+
+        @Argument(help: "URL of the collection to remove")
+        var collectionUrl: String?
+
+        mutating func run() throws {
+            guard let collectionUrlString = collectionUrl, let collectionUrl = URL(string: collectionUrlString) else {
+                throw CollectionsError.missingArgument("collectionUrl")
+            }
+
+            let collections = try await { self.collections.listCollections(identifiers: nil, in: profileOptions.usedProfile, callback: $0) }
+            let source = PackageCollectionsModel.CollectionSource(url: collectionUrl)
+
+            guard let collection = collections.first(where: { $0.source == source }) else {
+                throw CollectionsError.noCollectionMatchingURL(collectionUrlString)
+            }
+
+            _ = try await { self.collections.removeCollection(source, from: profileOptions.usedProfile, callback: $0) }
+            print("Removed \"\(collection.name)\" from your package collections.")
+        }
+    }
+
+    struct DescribeCollection: ParsableCommand {
+        static let configuration = CommandConfiguration(abstract: "Get metadata for a configured collection")
+
+        @Argument(help: "URL of the collection to describe")
+        var collectionUrl: String?
+
+        @OptionGroup
+        var profileOptions: ProfileOptions
+
+        mutating func run() throws {
```

Comment:
This can use the `getCollection` API--it fetches locally first and fetches the remote collection if not found

Reply:
+1 that was the original intent I believe